File: spoj/codechef/international_coding_maranthon/2.cpp
#include<bits/stdc++.h>
using namespace std;

int main(){

    int t;
    cin>>t;
    while(t--){
        int a,b,c,d;cin>>a>>b>>c>>d;
        if(abs(a-b) % abs(c-d) == 0){
            cout<<"YES\n";
        }
        else{
            cout<<"NO\n";
        }
    }
}

File: spoj/codechef/international_coding_maranthon/2.py
from sys import stdin,stdout
take = lambda: list(map(int,stdin.readline().split()))
mod = 1000000007
def power(n,m):
    res = 1
    while m >0:
        if n&1:
            res = res*n%mod
        m>>=1
        n = n*n%mod
    return res


for _ in range(int(input())):
    a,b,c,d, = take()
    if abs(a-b)%abs(c-d) == 0:
        print("YES")
    else:
        print("NO")
